src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/ViewContext.cs

```diff
@@ -58,16 +58,6 @@ public ViewContext()
                 throw new ArgumentNullException(nameof(view));
             }
 
-            if (viewData == null)
```

Comment:
@dougbu ah that's interesting. I suppose it may very well be better to have dummy ViewData/TempData for those cases. When I talked to @ryanbrandenburg I was indeed thinking only of explicit calls to ViewData/TempData from the view code, which I would be fine with failing.